### ISSUE TITLE
Assertion failed in ProjectRepositoryValidator.checkSyncStashState

### ISSUE THREAD

**Issue by matthewmichihara** [GoogleCloudPlatform/cloud-code-intellij#2915]
Cloud Code 21.3.1-202
Windows 10
IntelliJ IDEA 2021.1

```
java.lang.Throwable: Assertion failed
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:197)
	at com.intellij.openapi.diagnostic.Logger.assertTrue(Logger.java:206)
	at org.jetbrains.ide.BuiltInServerManagerImpl.waitForStart(BuiltInServerManagerImpl.kt:107)
	at org.jetbrains.git4idea.ssh.GitXmlRpcHandlerService.getXmlRcpPort(GitXmlRpcHandlerService.java:68)
	at git4idea.commands.GitHandlerAuthenticationManager.prepareHttpAuth(GitHandlerAuthenticationManager.java:93)
	at git4idea.commands.GitHandlerAuthenticationManager.lambda$prepare$0(GitHandlerAuthenticationManager.java:60)
	at git4idea.GitUtil.tryRunOrClose(GitUtil.java:1040)
	at git4idea.commands.GitHandlerAuthenticationManager.prepare(GitHandlerAuthenticationManager.java:59)
	at git4idea.commands.GitImplBase.run(GitImplBase.java:169)
	at git4idea.commands.GitImplBase.runCommand(GitImplBase.java:71)
	at git4idea.changes.GitChangeUtils.resolveReference(GitChangeUtils.java:160)
	at com.google.cloud.tools.intellij.cloud.debugger.ProjectRepositoryValidator.checkSyncStashState(ProjectRepositoryValidator.java:179)
	at com.google.cloud.tools.intellij.cloud.debugger.ui.CloudAttachDialog.checkSyncStashState(CloudAttachDialog.java:283)
	at com.google.cloud.tools.intellij.cloud.debugger.ui.CloudAttachDialog.lambda$new$1(CloudAttachDialog.java:145)
	at java.desktop/javax.swing.JComboBox.fireActionEvent(JComboBox.java:1264)
	at java.desktop/javax.swing.JComboBox.contentsChanged(JComboBox.java:1337)
	at java.desktop/javax.swing.AbstractListModel.fireContentsChanged(AbstractListModel.java:123)
	at java.desktop/javax.swing.DefaultComboBoxModel.setSelectedItem(DefaultComboBoxModel.java:94)
	at java.desktop/javax.swing.DefaultComboBoxModel.addElement(DefaultComboBoxModel.java:132)
	at java.desktop/javax.swing.JComboBox.addItem(JComboBox.java:716)
	at com.google.cloud.tools.intellij.cloud.debugger.ui.ProjectDebuggeeBinding.lambda$refreshDebugTargetList$0(ProjectDebuggeeBinding.java:211)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:969)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:839)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:797)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:781)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:502)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1063)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:694)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:436)
	at com.intellij.openapi.ui.DialogWrapper.doShow(DialogWrapper.java:1702)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1661)
	at git4idea.DialogManager.showDialog(DialogManager.java:62)
	at git4idea.DialogManager.show(DialogManager.java:28)
	at com.google.cloud.tools.intellij.cloud.debugger.CloudDebuggerRunner$1.start(CloudDebuggerRunner.java:107)
	at com.intellij.xdebugger.impl.XDebuggerManagerImpl.startSession(XDebuggerManagerImpl.java:270)
	at com.intellij.xdebugger.impl.XDebuggerManagerImpl.startSession(XDebuggerManagerImpl.java:231)
	at com.google.cloud.tools.intellij.cloud.debugger.CloudDebuggerRunner.createContentDescriptor(CloudDebuggerRunner.java:90)
	at com.google.cloud.tools.intellij.cloud.debugger.CloudDebuggerRunner.doExecute(CloudDebuggerRunner.java:74)
	at com.intellij.execution.runners.DefaultProgramRunner$execute$1.invoke(DefaultProgramRunner.kt:20)
	at com.intellij.execution.runners.DefaultProgramRunner$execute$1.invoke(DefaultProgramRunner.kt:14)
	at com.intellij.execution.impl.ExecutionManagerImpl$startRunProfile$1.invoke(ExecutionManagerImpl.kt:168)
	at com.intellij.execution.impl.ExecutionManagerImpl$startRunProfile$1.invoke(ExecutionManagerImpl.kt:68)
	at com.intellij.execution.impl.ExecutionManagerImpl$doStartRunProfile$startRunnable$1.run(ExecutionManagerImpl.kt:229)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:218)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:200)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:781)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:319)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:84)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:133)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:189)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:969)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:839)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:808)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:781)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:496)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
	```

**Comment by etanshaul** [comment]
another hit: https://listnrcrash.corp.google.com/product/717949/report/1000001068247658112

**Comment by etanshaul** [comment]
another one: https://listnrcrash.corp.google.com/product/717949/report/1000001061309904131

**Comment by peterlin741** [comment]
Fixed in `21.7.1`
